Library/Homebrew/utils/curl.rb

```diff
@@ -290,7 +290,25 @@ def curl_check_http_content(url, url_type, specs: {}, user_agents: [:default],
           url_protected_by_cloudflare?(response) || url_protected_by_incapsula?(response)
         end
 
-        return "The #{url_type} #{url} is not reachable (HTTP status code #{details[:status_code]})"
+        # https://github.com/Homebrew/brew/issues/13789
+        # If the `:homepage` of a formula is private, it will fail an `audit`
+        # since there's no way to specify a `strategy` with `using:` and
+        # Github does not authorize access to the web ui using token
+        #
+        # Strategy:
+        # If the `:homepage` 404s, it's a github link, and we have a token--
+        # check the API for the repository existing (which does use tokens)
```

Comment:
```suggestion
        # check the API for the repository (which does use tokens)
```

Reply:
Also moved parenthetical next to "API" to be more clear that it's the API that uses tokens, not the repository. Let me know if that's alright